lib/sqlalchemy/util/compat.py

```diff
@@ -327,12 +302,32 @@ def nested(*managers):
                 exc = sys.exc_info()
         if exc != (None, None, None):
             reraise(exc[0], exc[1], exc[2])
+    exc = (None, None, None)
```

Comment:
this line moving from up above is what broke the tests in https://jenkins.sqlalchemy.org/job/sqlalchemy_gerrit/cext=cext,db=sqlite-postgresql,pyv=py27/1930/#showFailuresLink, can you move that back?   you can re-push to this PR and I should be able to continue from there.